src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -0,0 +1,160 @@
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The module for processing proof nodes into Alethe proof nodes
+ */
+
+#include "proof/alethe/alethe_post_processor.h"
+
+#include "expr/node_algorithm.h"
+#include "proof/proof.h"
+#include "proof/proof_checker.h"
+#include "util/rational.h"
+
+namespace cvc5 {
+
+namespace proof {
+
+// This function removes all attributes contained in a given list of attributes
+// from a Node res while only recursively updating the node further if
+// continueRemoval is true.
+static Node removeAttributes(Node res,
+                             const std::vector<Kind>& attributes,
+                             bool (*continueRemoval)(Node))
+{
+  if (res.getNumChildren() != 0)
+  {
+    std::vector<Node> new_children;
+    if (res.hasOperator())
+    {
+      new_children.push_back(res.getOperator());
+    }
+    for (Node r : res)
+    {
+      if (std::find(attributes.begin(), attributes.end(), r.getKind())
+          == attributes.end())
+      {
+        new_children.push_back(
+            proof::removeAttributes(r, attributes, continueRemoval));
+      }
+    }
+    return NodeManager::currentNM()->mkNode(res.getKind(), new_children);
+  }
+  return res;
+}
+
+AletheProofPostprocessCallback::AletheProofPostprocessCallback(
+    ProofNodeManager* pnm)
+    : d_pnm(pnm)
+{
+  NodeManager* nm = NodeManager::currentNM();
+  d_cl = nm->mkBoundVar("cl", nm->stringType());
+}
+
+bool AletheProofPostprocessCallback::shouldUpdate(std::shared_ptr<ProofNode> pn,
+                                                  const std::vector<Node>& fa,
+                                                  bool& continueUpdate)
+{
+  return pn->getRule() != PfRule::ALETHE_RULE;
+}
+
+bool AletheProofPostprocessCallback::update(Node res,
+                                            PfRule id,
+                                            const std::vector<Node>& children,
+                                            const std::vector<Node>& args,
+                                            CDProof* cdp,
+                                            bool& continueUpdate)
+{
+  Trace("alethe-proof") << "- Alethe post process callback " << res << " " << id
+                        << " " << children << " / " << args << std::endl;
+
+  NodeManager* nm = NodeManager::currentNM();
+  std::vector<Node> new_args = std::vector<Node>();
+
+  switch (id)
+  {
+    //================================================= Core rules
+    //======================== Assume and Scope
+    // ======== Assumption (a leaf)
+    // Children: none
+    // Arguments: (F)
+    // --------------
+    // Conclusion: F
+    //
+    // proof rule: assume
+    // proof node: (VP:F)
+    // proof term: F
+    // premises: ()
+    // args: ()
```

Comment:
I'd prefer to delete this. I understand it's helpful during development, but now that `addAletheStep` is well documented this conversion is pretty straightforward without the above comment.

Reply:
I would vote to keep these comments since, e.g. in the example of SCOPE, it might be confusing to have an artificially constructed proof node that is created to print the same proof node differently. Also sometimes the proof node and term are hard to read in the code.

I will still delete the comments.

Reply:
I agree that there are cases (like for #7224) where it's certainly helpful to have such comments, but they can be written in a way where it's clearer what's going on, as judgments, like here: https://github.com/cvc5/cvc5/blob/c07d80a0ce4dc7144bad7146b0dc96574dd250c9/src/theory/uf/eq_proof.h#L107